src/AppInstallerCLICore/Workflows/WorkflowBase.h

```diff
@@ -197,6 +207,24 @@ namespace AppInstaller::CLI::Workflow
     // Inputs: SearchResult
     // Outputs: InstalledPackageVersion
     void GetInstalledPackageVersion(Execution::Context& context);
+
+    // Searches the source for the specific field as a completion.
+    // Required Args: None
+    // Inputs: CompletionData, Source
+    // Outputs: None
+    struct ReportExecutionStage : public WorkflowTask
+    {
+        ReportExecutionStage(ExecutionStage stage, bool allowBackward = false) : WorkflowTask("ReportExecutionStage"), m_stage(stage), m_allowBackward(allowBackward) {}
+
+        void operator()(Execution::Context& context) const override;
+
+    private:
+        mutable ExecutionStage m_stage;

```

Comment:
>mutable [](start = 8, length = 7)

This doesn't need mutable; you aren't writing to it in a const context. (And the move you are doing is completely unnecessary)

Reply:
Yea, but the compiler is not happy if I don't use std::move() since the context.Add() only takes rvalue.

---
In reply to: [508324457](https://github.com/microsoft/winget-cli/pull/617#discussion_r508324457) [](ancestors = 508324457)

Reply:
Then fix Add to have an lvalue version.

---
In reply to: [508751833](https://github.com/microsoft/winget-cli/pull/617#discussion_r508751833) [](ancestors = 508751833,508324457)